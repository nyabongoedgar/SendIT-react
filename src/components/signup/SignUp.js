import React from "react";
import { BrowserRouter, Link } from 'react-router-dom';
import LoginContainer from '../../containers/login/LoginContainer';

const SignUp = props => {
  let Loader = require('react-loader')
  return (
    <div className="authForms">
      <form onSubmit={props.submitHandler} id="#signUpForm">
      <h1>SIGN UP</h1>
        <input
          value={props.username}
          onChange={props.inputHandler}
          required
          name="username"
          id="username"
          placeholder="Username"
          type="text"
        />
        <br />
        <input
          value={props.email}
          onChange={props.inputHandler}
          required
          name="email"
          id="email"
          placeholder="Email Address"
          type="text"
        />
        <br />
        <input
          value={props.password}
          onChange={props.inputHandler}
          required
          name="password"
          id="password"
          placeholder="Password"
          type="password"
        />
        <br />
        <input
          value={props.password2}
          onChange={props.inputHandler}
          required
          name="password2"
          id="password2"
          placeholder="Confirm Password"
          type="password"
        />
        <br />
        <Loader loaded={!props.isLoading}>
        <button type="submit" className="btn regBtn">
          Create account
        </button>
        </Loader>
        <br />
          I have an account! 
          <Link to="/login">Login</Link>
        
        <br />
      </form>
    </div>
  );
};

export default SignUp;
