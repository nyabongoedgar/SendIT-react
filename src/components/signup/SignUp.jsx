import React from "react";

const SignUp = props => {
  return (
    <div>
      <form onSubmit={props.submitHandler}>
        <input
          value={props.username}
          onChange={props.inputHandler}
          required=""
          name="username"
          id="username"
          placeholder="Username"
          type="text"
        />
        <br />
        <input
          value={props.email}
          onChange={props.inputHandler}
          required=""
          name="email"
          id="email"
          placeholder="Email Address"
          type="text"
        />
        <br />
        <input
          value={props.password}
          onChange={props.inputHandler}
          required=""
          name="password"
          id="password"
          placeholder="Password"
          type="password"
        />
        <br />
        <input
          value={props.password2}
          onChange={props.inputHandler}
          required=""
          name="password2"
          id="password2"
          placeholder="Confirm Password"
          type="password"
        />
        <br />
        <button type="submit" className="btn">
          Create account
        </button>
        <br />
        <p id="error">{props.error}</p> <br />
        <p>
          I have an account! <a href="home.html">Login</a>
        </p>
        <br />
      </form>
    </div>
  );
};

export default SignUp;
