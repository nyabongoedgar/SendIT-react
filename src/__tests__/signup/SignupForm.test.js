import React from 'react';
import SignUp from '../../components/signup/SignUp';
import renderer from 'react-test-renderer';


describe('SignupForm Component', () => {

  it('it matches the snapshot', () => {
    const tree = renderer.create(<SignUp />).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
