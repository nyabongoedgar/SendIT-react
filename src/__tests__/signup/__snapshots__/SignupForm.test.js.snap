// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm Component it matches the snapshot 1`] = `
<div
  className="authForms"
>
  <form
    id="#signUpForm"
  >
    <h1>
      SIGN UP
    </h1>
    <input
      id="username"
      name="username"
      placeholder="Username"
      required={true}
      type="text"
    />
    <br />
    <input
      id="email"
      name="email"
      placeholder="Email Address"
      required={true}
      type="text"
    />
    <br />
    <input
      id="password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <br />
    <input
      id="password2"
      name="password2"
      placeholder="Confirm Password"
      required={true}
      type="password"
    />
    <br />
    <div
      className="loadedContent"
    >
      <button
        className="btn regBtn"
        type="submit"
      >
        Create account
      </button>
    </div>
    <br />
    I have an account! 
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <br />
  </form>
</div>
`;
